src/rp2040.ts

```diff
@@ -350,31 +372,52 @@ export class RP2040 {
   }
 
   step() {
-    this.core.executeInstruction();
+    this.core0.executeInstruction();
+    this.core1.executeInstruction();
   }
 
   execute() {
+    this.core0.stopped = false;
+    this.core1.stopped = false;
+    setTimeout(() => this.executeInternal(), 0);
+  }
+
+  private executeInternal() {
     this.clock.resume();
     this.executeTimer = null;
-    this.stopped = false;
-    for (let i = 0; i < 100000 && !this.stopped && !this.core.waiting; i++) {
-      this.core.executeInstruction();
-    }
-    if (!this.stopped) {
-      this.executeTimer = setTimeout(() => this.execute(), 0);
+    let idle = false;
+    for (let i = 0; i < 1000 && !idle; i++) {
+      idle = true;
+      if (!this.core0.stopped && !this.core0.waiting) {
+        idle = false;
+        this.isCore0Running = true;
+        this.core0.executeInstruction();
+      }
+      if (!this.core1.stopped && !this.core1.waiting) {
+        idle = false;
+        this.isCore0Running = false;
+        this.core1.executeInstruction();
+      }
     }
+    this.executeTimer = setTimeout(() => this.executeInternal(), 0);
   }
 
   stop() {
-    this.stopped = true;
+    this.core0.stopped = true;
```

Comment:
As far as I understand, `stopped` is a bit of a kludge for controlling the emulation machinery (that's running via `setTimeout()`) from outside code; it's not a feature of the actual RP2040 hardware. Since the cores are not running as callbacks individually, there's probably no need to give each core a `stopped` attribute.

That said, I'd just remove the `setTimeout()` stuff from rp2040js altogether; it seems to be some leftover from Wokwi. It'd probably be better to just have a `rp2040.step()` function and leave everything else to the calling code. (but this might be outside the scope of this PR)

Reply:
The `stopped` flag for each core is necessary. Take [pico-debug](https://github.com/majbthrd/pico-debug) as an example, it use one core to debug another core.

Reply:
The pico-debug project uses the SYSCFG DBGFORCE register, basically rerouting the SWD functionality built into the (real, hardware) Cortex cores via USB. This is unrelated to `stopped`, and SWD functionality isn't implemented at all in rp2040js.

Reply:
What I want to say is the 2 cores may be 1 running and 1 stopped. And you are right, pico-debug use SWD which is not related to our `stopped` flag. But you can check gdb server in this project which only stop core0 while debugging.